internal/configs/test_file.go

```diff
@@ -324,6 +326,18 @@ type TestRunOptions struct {
 	DeclRange hcl.Range
 }
 
+// runBlockBackend is used when parsing a single test file as part of ensuring
+// there is only a single backend block for a given internal state file.
+// TODO(SarahFrench) this also needs to be done across multiple test files - ?
```

Comment:
I need to check if parallelism work is within the scope of an individual file, or in the scope of an entire test suite.

Reply:
parallelism works within the scope of a file. in addition, runs with the same state file are never executed in parallel, and that will remain true for runs with backend

Reply:
That's good! Identifying if the same backend is used across multiple files would be a tough thing to detect when validating configuration. If files are always sequential then we can allow users to re-use preexisting state between test files if they want/need, and if they experienced any issues it'd be simple enough for them to debug themselves.